client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -587,6 +589,9 @@ private boolean revive(
               ClassTag$.MODULE$.apply(PbChangeLocationResponse.class));
       // per partitionKey only serve single PartitionLocation in Client Cache.
       StatusCode respStatus = Utils.toStatusCode(response.getStatus());
+      if (!response.getExcluded()) {
```

Comment:
Why not remove newLocation worker from blacklist, And IMO client side blacklist would sometimes more timely then LifecycleManager, so it maybe not credible from LM.

Reply:
The newLocation should not be in executor's excluded list because revive will exclude the oldPartition. Executor's excluded list will be added in LifecycleManager's excluded list, see ```ChangePartitionManager::handleRequestPartitions```. Currently, LM has a more complex and accurate strategy to mange its excluded list, so I think it's more accurate than executor side.